test/e2e/storage/non_csi_volumes.go

```diff
@@ -0,0 +1,1567 @@
+	"volumeIO":                         initVolumeIOTestCase,
+	"volumeMode [Feature:BlockVolume]": initVolumeModeTestCase,
+	"subPath":                          initSubPathTestCase,
+}
+
+var _ = utils.SIGDescribe("Non-CSI", func() {
+	f := framework.NewDefaultFramework("volumes")
+
+	var (
+		cs     clientset.Interface
+		ns     *v1.Namespace
+		node   v1.Node
+		config framework.VolumeTestConfig
+	)
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace
+		nodes := framework.GetReadySchedulableNodesOrDie(cs)
+		node = nodes.Items[rand.Intn(len(nodes.Items))]
+		config = framework.VolumeTestConfig{
+			Namespace:      ns.Name,
+			Prefix:         "volume",
+			ClientNodeName: node.Name,
+			ServerNodeName: node.Name,
+		}
+	})
+
+	for testCaseName, testCaseInit := range testCases {
+		curTestCaseName := testCaseName
+		curTestCase := testCaseInit()
+		testPatterns := curTestCase.getTestPatterns()
+		curTestFunc := curTestCase.getTestFunc()
+
+		for testPatternName, testPattern := range testPatterns {
+			curTestPatternName := testPatternName
+			curTestPattern := testPattern
+
+			for driverName, initDriver := range testDrivers {
+				curDriverName := driverName
+				curInitDriver := initDriver
+				Context(fmt.Sprintf("Driver: %s, Test: %s, TestPattern: %s",
+					curDriverName, curTestCaseName, curTestPatternName), func() {
+					var (
+						driver       testDriver
+						testCase     testCase
+						testResource testResource
+						testInput    testInput
+					)
+
+					BeforeEach(func() {
+						driver = curInitDriver(f, config, node)
+						// Skip tests before driver creates test resource to avoid unnecessary initialization
+						skipTestForDriver(curTestPattern, driver)
+
+						driver.createDriver()
+						testCase = curTestCase
+						testResource = testCase.initTestResource()
+						testResource.setup(driver, curTestPattern)
+						testInput = testCase.createTestInput(curTestPattern, testResource)
+					})
+
+					AfterEach(func() {
+						testResource.cleanup()
+						driver.cleanupDriver()
+					})
+
+					curTestFunc(&testInput)
+				})
+			}
+		}
+	}
+})
+
+type testDriver interface {
+	createDriver()
+	cleanupDriver()
+	getDriverInfo() driverInfo
+}
+
+type inlineVolumeTestDriver interface {
+	testDriver
+	getVolumeSource(readOnly bool, fsType string) *v1.VolumeSource
+}
+
+type staticPVTestDriver interface {
+	testDriver
+	getPersistentVolumeSource(readOnly bool, fsType string) *v1.PersistentVolumeSource
+}
+
+type dynamicPVTestDriver interface {
+	testDriver
+	getDynamicProvisionStorageClass(fsType string) *storagev1.StorageClass
+}
+
+type driverInfo struct {
+	name                      string
+	fsGroup                   *int64
+	podSec                    v1.PodSecurityContext
+	privilegedSecurityContext bool
+	maxFileSize               int64
+
+	// supportablity for driver
+	supportedFsType  map[string]bool
+	isBlockSupported bool
+
+	// volumes test specific parameters
```

Comment:
why do these two parameters need to be set by the driver?  Can you generate a "expected content" randomly based off of test namespace?

Reply:
Make testSuite always generate content.

My intention was to keep compatibility from the exsiting tests, which prepare volumes with contents.
However, it won't be a must and I agree to change it as suggested to make ```driverInfo``` simple and clean.